SantasBag/SantasBag.DataAccess/Repositories/RewardsRepository.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SantasBag.Core.Models;
+using SantasBug.Core.Abstractions;
+using SantasBug.DataAccess.Entities;
+
+namespace SantasBug.DataAccess.Repositories;
+
+public class RewardsRepository : IRewardsRepository
+{
+    private readonly SantasBagDbContext _context;
+
+    public RewardsRepository(SantasBagDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<List<Reward>> Get(CancellationToken cancellationToken)
+    {
+        var rewardEntities = await _context.Rewards
+            .AsNoTracking()
+            .ToListAsync();
+        var rewards = rewardEntities
+            .Select(b=>Reward.Create(b.Name, b.Description, b.Image, b.Cost, b.RoomId).Reward)
+            .ToList();
+        return rewards;
+    }
+
+    public async Task<Guid> Create(Reward reward, CancellationToken cancellationToken)
+    {
```

Comment:
Furthermore, your methods in repositories have to use your Entities model instead of business models, because repositories is directly related with DataAccess layer, so DataAccess => Entities and BusinessLogic => Dto's models